nautobot/utilities/tests/test_utils.py

```diff
@@ -346,3 +355,173 @@ def test_slugify_dashes_to_underscores(self):
             ("alpha-beta_gamma delta", "alpha_beta_gamma_delta"),
         ):
             self.assertEqual(slugify_dashes_to_underscores(content), expected)
+
+
+class LookupRelatedFunctionTest(TestCase):
+    def test_is_single_choice_field(self):
+        # Assert function returns True for any field starting with create or has_
+        # Cause these fields are either boolean fields or date time fields which one accepts single values
+        filterset_class = SiteFilterSet
+
+        single_choice_fields = ("created", "created__gte", "has_vlans", "has_clusters")
+        for field in single_choice_fields:
+            self.assertTrue(is_single_choice_field(filterset_class, field))
+
+        multi_choice_fields = ("status", "tenant", "tag")
+        for field in multi_choice_fields:
+            self.assertFalse(is_single_choice_field(filterset_class, field))
+
+    def test_build_lookup_label(self):
+        with self.subTest():
+            label = build_lookup_label("slug__iew", "iendswith")
```

Comment:
That's the verbose name for `iew`, which is needed by `django_filters.utils.verbose_lookup_expr(_verbose_name)` to get the user-friendly version -> `ends-with`.

Reply:
And there's no straightforward way for us to go from `slug__iew` --> `iew` --> `iendswith` to avoid needing to pass both `slug__iew` and `iendswith` as distinct parameters?

Reply:
Conflict may arise. An example I can think of is:
`status__n` -> `n` -> `exact`
`tenantgroup__n` -> `n` -> `in`

How do i figure out which is `in` and which is `exact`

Reply:
Currently the `verbose_name` is generated from `filterset_class().filters["status"].lookup_expr`